scalameta/parsers/shared/src/main/scala/scala/meta/internal/parsers/ScalametaParser.scala

```diff
@@ -2198,7 +2198,17 @@ class ScalametaParser(input: Input)(implicit dialect: Dialect) { parser =>
       val lpPos = auto.startTokenPos
       val args = checkNoTripleDots(argumentExprs(location))
       val rpPos = auto.endTokenPos
-      @inline def addPos(body: Term): Term = atPosWithBody(lpPos, body, rpPos)
+      @inline def addPos(body: Term): Term = {
+        body match {
+          // For `Term.Ref`, avoid using autoPos to exclude parens tokens in Term.ApplyInfix.args.
+          // For `a f (b)`, autoPos include `(b)` for ApplyInfix.args position.
+          // https://github.com/scalacenter/scalafix/issues/1594
+          case _: Term.Ref =>
```

Comment:
It is fine, as `(b + c)` part is something like `Term.ApplyInfix(lhs=b, op=+, targs=List(), rhs=List(c))` and `Term.ApplyInfix` is not a subclass of `Term.Ref`. :+1:
https://github.com/scalameta/scalameta/blob/1a9ec6f9aa06e20bc786b5c7ef6d7aa500e59eaf/scalameta/trees/shared/src/main/scala/scala/meta/Trees.scala#L91

Actually, that case is tested in https://github.com/scalameta/scalameta/blob/1a9ec6f9aa06e20bc786b5c7ef6d7aa500e59eaf/tests/shared/src/test/scala/scala/meta/tests/parsers/BasicPositionSuite.scala#L10-L14

Reply:
Shouldn't it be:
`Term.ApplyInfix 2 / 3 ` instead? We are adding the surrounding parenthesis here.

Anyway, it's still an improvement, so we can merge as is. The biggest issue was with the refs I guess

Reply:
> Term.ApplyInfix 2 / 3 instead? We are adding the surrounding parenthesis here.

Hmm, indeed it would be better.
However, the problem is, if we exclude the parens from `Term.ApplyInfix`, `tree.syntax` will show `1 + 2 / 3` which is broken.

Reply:
> However, the problem is, if we exclude the parens from Term.ApplyInfix, tree.syntax will show 1 + 2 / 3 which is broken.

shouldn't parens always be excluded and conditionally added in the pretty-printer then?

Reply:
@tanishiking are you observing syntax `1 + 2 / 3` for the entire expression, missing the `* 4` at the end?

Reply:
@tanishiking are you observing syntax `1 + 2 / 3` for the entire expression, missing the `* 4` at the end?